src/test/java/org/jitsi/jicofo/ParticipantTest.java

```diff
@@ -44,88 +42,397 @@
-
-        MockRoomMember roomMember
-            = mockMultiUserChat.createMockRoomMember("testMember");
-
-        Participant participant
-            = new Participant(mockConference, roomMember, maxSSRCCount);
-
-        SourcePacketExtension[] audioSSRCs
-            = new SourcePacketExtension[] {
-            SSRCUtil.createSSRC(1L, new String[][]{
-                {"cname", "cname1"},
-                {"msid", "stream1 track1"}
-            }),
-            SSRCUtil.createSSRC(2L, new String[][]{
-                {"cname", "cname2"},
-                {"msid", "stream2 track2"}
-            }),
-            // Duplicated SSRC
-            SSRCUtil.createSSRC(1L, new String[][]{
-                {"cname", "cname3"},
-                {"msid", "stream3 track3"}
-            }),
-            // Duplicated media stream id
-            SSRCUtil.createSSRC(3L, new String[][]{
-                {"cname", "cname2"},
-                {"msid", "stream2 track2"}
-            }),
-            SSRCUtil.createSSRC(4L, new String[][]{
-                {"cname", "cname4"},
-                {"msid", "stream4 track4"}
-            })};
+    private MockJitsiMeetConference mockConference;
+    private MockRoomMember roomMember;
+    private Participant participant;
+
+    private ArrayList<ContentPacketExtension> answerContents;
+    private RtpDescriptionPacketExtension audioRtpDescPe;
+    private RtpDescriptionPacketExtension videoRtpDescPe;
+
+    private SourcePacketExtension[] audioSSRCs;
+
+    private SourcePacketExtension[] videoSSRCs;
+    private SourceGroupPacketExtension[] videoGroups;
+
+    @Before
+    public void setUpContents()
+    {
+        this.mockConference = new MockJitsiMeetConference();
+        MockMultiUserChat mockMultiUserChat = new MockMultiUserChat(null, null);
+        this.roomMember = mockMultiUserChat.createMockRoomMember("testMember");
+        this.participant
+            = new Participant(mockConference, roomMember, 20);
+
+        SourcePacketExtension ssrc1 = createSSRC(1L);
+        SourcePacketExtension ssrc2 = createSSRC(2L);
+        SourcePacketExtension ssrc4 = createSSRC(4L);
+
+        this.audioSSRCs = new SourcePacketExtension[] { ssrc1, ssrc2, ssrc4};
+
+        String cname = "videocname";
+        String msid = "vstream vtrack";
+
+        this.videoSSRCs = new SourcePacketExtension[] {
+            createGroupSSRC(10L, cname, msid),
+            createGroupSSRC(20L, cname, msid),
+            createGroupSSRC(30L, cname, msid),
+            createGroupSSRC(40L, cname, msid),
+            createGroupSSRC(50L, cname, msid),
+            createGroupSSRC(60L, cname, msid)
+        };
+
+        this.videoGroups = new SourceGroupPacketExtension[] {
+            createGroup(
+                SourceGroupPacketExtension.SEMANTICS_SIMULCAST,
+                new long[] { 10L, 30L, 50L },
+                cname, msid),
+            createGroup("FIR", new long[] { 10L, 20L }, cname, msid),
```

Comment:
It would probably make sense to add constant to SourceGroupPacketExtension as well, but then I have to go and update protocol-jabber... do you think I should do that now ?